superset/db_engine_specs/starrocks.py

```diff
@@ -0,0 +1,239 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# pylint: no-init
+import logging
+import re
+from typing import Any, Dict, List, Optional, Pattern, Tuple, Type
+from urllib import parse
+
+from flask_babel import gettext as __
+from sqlalchemy import Integer, Numeric, types
+from sqlalchemy.engine import Inspector
+from sqlalchemy.engine.result import Row as ResultRow
+from sqlalchemy.engine.url import URL
+from sqlalchemy.sql.type_api import TypeEngine
+
+from superset.db_engine_specs.mysql import MySQLEngineSpec
+from superset.errors import SupersetErrorType
+from superset.utils.core import GenericDataType
+
+# Regular expressions to catch custom errors
+CONNECTION_ACCESS_DENIED_REGEX = re.compile(
+    "Access denied for user '(?P<username>.*?)'"
+)
+CONNECTION_UNKNOWN_DATABASE_REGEX = re.compile("Unknown database '(?P<database>.*?)'")
+
+logger = logging.getLogger(__name__)
+
+
+class TINYINT(Integer):
+    __visit_name__ = "TINYINT"
+
+
+class DOUBLE(Numeric):
+    __visit_name__ = "DOUBLE"
+
+
+class ARRAY(TypeEngine):
+    __visit_name__ = "ARRAY"
+
+    @property
+    def python_type(self) -> Optional[Type[List[Any]]]:
+        return list
+
+
+class MAP(TypeEngine):
+    __visit_name__ = "MAP"
+
+    @property
+    def python_type(self) -> Optional[Type[Dict[Any, Any]]]:
+        return dict
+
+
+class STRUCT(TypeEngine):
+    __visit_name__ = "STRUCT"
+
+    @property
+    def python_type(self) -> Optional[Type[Any]]:
+        return None
+
+
+class StarRocksEngineSpec(MySQLEngineSpec):
+    engine = "starrocks"
+    engine_name = "StarRocks"
+
+    default_driver = "starrocks"
+    sqlalchemy_uri_placeholder = (
+        "starrocks://user:password@host:port/catalog.db[?key=value&key=value...]"
+    )
+
+    _default_column_type_mappings = (
```

Comment:
this should be 
```suggestion
    column_type_mappings = (
```

Also note, that you don't need to reimplement all types - Any types that are already covered by `_default_column_type_mappings` can be omitted. In addition, I would suggest adding unit tests to ensure that the main types are identified correctly (see `tests/unit_tests/db_engine_specs` for other examples of how this is done).

Reply:
thanks:)
added unit tests and updated codes, please review it again.